engine/apps/api/permissions/__init__.py

```diff
@@ -1,5 +1,196 @@
+        CHATOPS_WRITE = _generate_permission_string(Resources.CHATOPS, Actions.WRITE)
+
+        OUTGOING_WEBHOOKS_READ = _generate_permission_string(Resources.OUTGOING_WEBHOOKS, Actions.READ)
+        OUTGOING_WEBHOOKS_WRITE = _generate_permission_string(Resources.OUTGOING_WEBHOOKS, Actions.WRITE)
+
+        MAINTENANCE_READ = _generate_permission_string(Resources.MAINTENANCE, Actions.READ)
+        MAINTENANCE_WRITE = _generate_permission_string(Resources.MAINTENANCE, Actions.WRITE)
+
+        API_KEYS_READ = _generate_permission_string(Resources.API_KEYS, Actions.READ)
+        API_KEYS_WRITE = _generate_permission_string(Resources.API_KEYS, Actions.WRITE)
+
+        NOTIFICATION_SETTINGS_READ = _generate_permission_string(Resources.NOTIFICATION_SETTINGS, Actions.READ)
+        NOTIFICATION_SETTINGS_WRITE = _generate_permission_string(Resources.NOTIFICATION_SETTINGS, Actions.WRITE)
+
+        USER_SETTINGS_READ = _generate_permission_string(Resources.USER_SETTINGS, Actions.READ)
+        USER_SETTINGS_WRITE = _generate_permission_string(Resources.USER_SETTINGS, Actions.WRITE)
+        USER_SETTINGS_ADMIN = _generate_permission_string(Resources.USER_SETTINGS, Actions.ADMIN)
+
+        OTHER_SETTINGS_READ = _generate_permission_string(Resources.OTHER_SETTINGS, Actions.READ)
+        OTHER_SETTINGS_WRITE = _generate_permission_string(Resources.OTHER_SETTINGS, Actions.WRITE)
+
+        TESTING = _generate_permission_string(Resources._IMPOSSIBLE_TESTING, Actions._IMPOSSIBLE_TESTING)
+
+    def has_permission(self, request: Request, view: ViewSetOrAPIView) -> bool:
+        action = _get_view_action(request, view)
+
+        rbac_permissions: RBACPermissionsAttribute = getattr(view, RBAC_PERMISSIONS_ATTR, None)
+
+        assert (
+            rbac_permissions is not None
+        ), f"Must define a {RBAC_PERMISSIONS_ATTR} dict on the ViewSet that is consuming the RBACPermission class"
+
+        return has_permissions(request.user.permissions, rbac_permissions.get(action, []))
+
+    def has_object_permission(self, request: Request, view: ViewSetOrAPIView, obj: typing.Any) -> bool:
+        rbac_object_permissions: RBACObjectPermissionsAttribute = getattr(view, RBAC_OBJECT_PERMISSIONS_ATTR, None)
+
+        if rbac_object_permissions:
+            action = _get_view_action(request, view)
+
+            for PermissionClass, actions in rbac_object_permissions.items():
+                if action in actions:
+                    return PermissionClass.has_object_permission(request, view, obj)
+            return False
+
+        # has_object_permission is called after has_permission, so return True if in view there is not
+        # RBAC_OBJECT_PERMISSIONS_ATTR attr which mean no additional check involving object required
+        return True
+
+
+def has_permissions(
+    user_permissions: typing.List[GrafanaPermission], required_permissions: typing.List[RBACPermission.Permissions]
+) -> bool:
+    user_has_all_required_permissions = True
```

Comment:
Wondering if this could work and be more concise:

```python
user_permissions = [u["action"] for u in user_permissions]
required_permissions = [p.value for p in required_permissions]

return all(permission in user_permissions for permission in required_permissions)
```